src/ae.cpp

```diff
@@ -256,6 +269,21 @@ int aePostFunction(aeEventLoop *eventLoop, aePostFunctionProc *proc, void *arg)
     return AE_OK;
 }
 
+int aePostFunction(aeEventLoop *eventLoop, aePostFunctionTokenProc *proc, StorageToken *token)
+{
+  //added to support async api IStorage
+    proc(eventLoop,token);
```

Comment:
If you call it here then it defeats the purpose of writing it to the pipe (you will call it twice).  Please remove.

Reply:
Yes correct!. Also I think this aePostFunction, would also need function wrapper to run IStorage APIs ? 
int aePostFunction(aeEventLoop *eventLoop, aePostFunctionTokenProc *proc, StorageToken *token, std::function<StorageToken()> fn); . Please suggest